### PR TITLE
chore(amplify_api): REST integration tests and change API provision script to use init

### PR DESCRIPTION
**Pull request by ragingsquirrel3**
1) Add basic integration tests for REST API
2) Because these tests add requirements for the backend resource which CLI headless add cannot do, this PR also makes the API backend provision script use `amplify init` with a specific subset of the /amplify dir designed for this purpose.

2 is part of potentially larger migration to make all the integ tests use these backend files so it's easier to reproduce the stack, source control things like lambda code, and generally do the things headless CLI cannot do. amplify-ui uses [this strategy](https://github.com/aws-amplify/amplify-ui/tree/main/environments). See more documentation https://aws.amazon.com/blogs/mobile/amplify-cli-adds-scaffolding-support-for-amplify-apps-and-authoring-plugins/.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_api/example/integration_test/rest_tests.dart
Comment (dnys1) [inline packages/amplify_api/example/integration_test/rest_tests.dart:36]: Would remove this - probably want the tests to fail if configure doesn't execute.

File: packages/amplify_api/example/amplify/backend/function/flutterapiintegtest17270866/src/index.js
Comment (dnys1) [inline packages/amplify_api/example/amplify/backend/function/flutterapiintegtest17270866/src/index.js:12]: Can you return just the string? I was a bit confused why it was quoted in the test case - think a string makes more sense than a "JSON string"

File: packages/amplify_api/example/integration_test/rest_tests.dart
Comment (dnys1) [inline packages/amplify_api/example/integration_test/rest_tests.dart:24]: Can we add an error case? Could be 404 so there's no backend work to do.

**Review by dnys1 — APPROVED** [review]
Awesome! Thanks for doing this!

File: packages/amplify_api/example/integration_test/rest_tests.dart
Comment (ragingsquirrel3) [reply packages/amplify_api/example/integration_test/rest_tests.dart:36]: Want to keep to support running the file individually (rather than from main test file).

File: packages/amplify_api/example/amplify/backend/function/flutterapiintegtest17270866/src/index.js
Comment (ragingsquirrel3) [reply packages/amplify_api/example/amplify/backend/function/flutterapiintegtest17270866/src/index.js]: updated. Good suggestion. This is how the CLI does the simple "hello world" lambda by default but agree it's confusing.

File: packages/amplify_api/example/integration_test/rest_tests.dart
Comment (ragingsquirrel3) [reply packages/amplify_api/example/integration_test/rest_tests.dart:24]: Good suggestion, added (although backend will actually give you 403 if you try some random route as I guess that's how API gateway is configured in this case).

File: packages/amplify_api/example/integration_test/rest_tests.dart
Comment (dnys1) [reply packages/amplify_api/example/integration_test/rest_tests.dart:36]: Just meant to remove the if condition - for example if it's already configured then the first test will fail but maybe not for the reasons we expect. We'd rather have a failure in the setup method if there are issues configuring.

File: packages/amplify_api/example/integration_test/rest_tests.dart
Comment (dnys1) [reply packages/amplify_api/example/integration_test/rest_tests.dart:24]: Perfect, works just as well. Thanks!

File: packages/amplify_api/example/integration_test/rest_tests.dart
Comment (ragingsquirrel3) [reply packages/amplify_api/example/integration_test/rest_tests.dart:36]: discussed offline, keeping as-is here

**Review by haverchuck — APPROVED** [review]
LGTM - just to confirm, we aren't checking in any api keys, right?

**Comment by ragingsquirrel3** [comment]
> LGTM - just to confirm, we aren't checking in any api keys, right?

No.

**Comment by ragingsquirrel3** [comment]
this CLI issue needs to be fixed before merging this or provision script won't work https://github.com/aws-amplify/amplify-cli/issues/9778

**Comment by ragingsquirrel3** [comment]
This issue https://github.com/aws-amplify/amplify-cli/issues/9508 is blocking merge as `amplify init` won't work until that's fixed.

**Pull request closed without merging**
